src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -0,0 +1,821 @@
+ *
+ * <p>These assertions are filesystem independent. You may use them on {@code
+ * Path} instances issued from the default filesystem (ie, instances you get
+ * when using {@link java.nio.file.Paths#get(String, String...)}) or from other
+ * filesystems. For more information, see the {@link FileSystem javadoc for
+ * {@code FileSystem}}.</p>
+ *
+ * <p>Furthermore:</p>
+ *
+ * <ul>
+ *     <li>Unless otherwise noted, assertions which accept arguments will not
+ *     accept {@code null} arguments; if a null argument is passed, these
+ *     assertions will throw a {@link NullPointerException}.</li>
+ *     <li>It is the caller's responsibility to ensure that paths used in
+ *     assertions are issued from valid filesystems which are not {@link
+ *     FileSystem#close() closed}. If a filesystems is closed, assertions will
+ *     throw a {@link ClosedFileSystemException}.</li>
+ *     <li>Some assertions take another {@link Path} as an argument. If this
+ *     path is not issued from the same {@link FileSystemProvider provider} as
+ *     the tested path, assertions will throw a {@link
+ *     ProviderMismatchException}.</li>
+ *     <li>Some assertions may need to perform I/O on the path's underlying
+ *     filesystem; if an I/O error occurs when accessing the filesystem, these
+ *     assertions will throw a {@link PathsException}.</li>
+ * </ul>
+ *
+ * @param <S> self type
+ *
+ * @see Path
+ * @see java.nio.file.Paths#get(String, String...)
+ * @see FileSystem
+ * @see FileSystem#getPath(String, String...)
+ * @see FileSystems#getDefault()
+ * @see Files
+ */
+public abstract class AbstractPathAssert<S extends AbstractPathAssert<S>>
+    extends AbstractAssert<S, Path>
+{
+    @VisibleForTesting
+    protected Paths paths = Paths.instance();
+
+    protected AbstractPathAssert(final Path actual, final Class<?> selfType)
+    {
+        super(actual, selfType);
+    }
+
+    /**
+     * Assert that a given path exists
+     *
+     * <p><strong>Note that this assertion will follow symbolic links before
+     * asserting the path's existence.</strong></p>
+     *
+     * <p>On Windows system, this has no influence. On Unix systems, this means
+     * the assertion result will fail if the path is a symbolic link whose
+     * target does not exist. If you want to assert the existence of the
+     * symbolic link itself, use {@link #existsNoFollow()} instead.</p>
+     *
+     * <p>Examples:</p>
+     *
+     * <pre><code class="java">
+     * // fs is a Unix filesystem
+     * // Create a symbolic link whose target does not exist
+     * final Path nonExistent = fs.getPath("nonexistent");
+     * final Path dangling = fs.getPath("dangling");
+     * Files.createSymbolicLink(dangling, nonExistent);
+     *
+     * // Create a regular file, and a symbolic link pointing to it
+     * final Path someFile = fs.getPath("somefile");
+     * Files.createFile(someFile);
+     * final Path symlink = fs.getPath("symlink");
```

Comment:
rename `symlink` to `someFileSymLink` to make it obvious what the link target is.